test/integration/scheduler/priorities_test.go

```diff
@@ -361,3 +362,105 @@ func TestEvenPodsSpreadPriority(t *testing.T) {
 		})
 	}
 }
+
+// TestDefaultPodTopologySpreadScore verifies that the PodTopologySpread Score plugin
+// with the system default spreading spreads Pods belonging to a Service.
+// The setup has 500 nodes over 3 zones.
+func TestDefaultPodTopologySpreadScore(t *testing.T) {
+	testCtx := initTestSchedulerForPriorityTest(t, podtopologyspread.Name)
+	t.Cleanup(func() {
+		testutils.CleanupTest(t, testCtx)
+	})
+	cs := testCtx.ClientSet
+	ns := testCtx.NS.Name
+
+	var nodes []*v1.Node
+	for i := 0; i < 500; i++ {
+		nodeName := fmt.Sprintf("node-%d", i)
+		zone := fmt.Sprintf("zone-%d", i%3)
+		node, err := createNode(cs, st.MakeNode().Name(nodeName).Label(v1.LabelHostname, nodeName).Label(v1.LabelZoneFailureDomainStable, zone).Obj())
+		if err != nil {
+			t.Fatalf("Cannot create node: %v", err)
+		}
+		nodes = append(nodes, node)
+	}
+
+	serviceName := "test-service"
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: ns,
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				"service": serviceName,
+			},
+			Ports: []v1.ServicePort{{
+				Port:       80,
+				TargetPort: intstr.FromInt(80),
+			}},
+		},
+	}
+	svc, err := cs.CoreV1().Services(ns).Create(testCtx.Ctx, svc, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Creating service: %v", err)
+	}
+
+	pause := imageutils.GetPauseImageName()
+	var pods []*v1.Pod
+
+	t.Cleanup(func() {
+		if err := cs.CoreV1().Services(ns).Delete(testCtx.Ctx, svc.Name, metav1.DeleteOptions{}); err != nil {
+			t.Errorf("Deleting service: %v", err)
+		}
+		testutils.CleanupPods(cs, t, pods)
+	})
+
+	for _, nPods := range []int{3, 9, 15} {
+		// Append nPods each iteration.
```

Comment:
So, the intention of "appending" pods is that for the 2nd iteration, we have 3 existing pods, and 12 existing pods on the 3rd iteration?

Reply:
Yeah. I initially intended to recreate all Pods from zero, but no need, as all the Pods are equal.